front/scripts/test_llm_clients.ts

```diff
@@ -0,0 +1,266 @@
+import type { ModelIdType, ReasoningEffort } from "@app/types";
+import type { ModelConversationTypeMultiActions } from "@app/types/assistant/generation";
+import {
+  CLAUDE_3_5_HAIKU_20241022_MODEL_ID,
+  CLAUDE_4_5_HAIKU_20251001_MODEL_ID,
+  CLAUDE_4_5_SONNET_20250929_MODEL_ID,
+} from "@app/types/assistant/models/anthropic";
+
+interface TestConfig {
+  provider: "anthropic";
+  modelId: ModelIdType;
+  temperature?: number;
+  reasoningEffort?: ReasoningEffort;
+}
+
+interface TestConversation {
+  name: string;
+  systemPrompt: string;
+  userMessage: string;
+  expectedInResponse?: string;
+}
+
+function generateAnthropicThinkingTestConfigs(
+  modelId: ModelIdType
+): TestConfig[] {
+  return [
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 0.5,
+      reasoningEffort: "none",
+    },
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 1,
+      reasoningEffort: "light",
+    },
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 0.5,
+      reasoningEffort: "medium",
+    },
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 1,
+      reasoningEffort: "high",
+    },
+  ];
+}
+
+function generateAnthropicNotThinkingTestConfigs(
+  modelId: ModelIdType
+): TestConfig[] {
+  return [
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 0.5,
+    },
+    {
+      provider: "anthropic",
+      modelId: modelId,
+      temperature: 1,
+    },
+  ];
+}
+
+const ANTHROPIC_NOT_THINKING_MODELS = [CLAUDE_3_5_HAIKU_20241022_MODEL_ID];
+
+const ANTHROPIC_THINKING_MODELS = [
+  CLAUDE_4_5_HAIKU_20251001_MODEL_ID,
+  CLAUDE_4_5_SONNET_20250929_MODEL_ID,
+];
+
+const TEST_CONFIGS: TestConfig[] = [
+  ...ANTHROPIC_THINKING_MODELS.flatMap(generateAnthropicThinkingTestConfigs),
+  ...ANTHROPIC_NOT_THINKING_MODELS.flatMap(
+    generateAnthropicNotThinkingTestConfigs
+  ),
+];
+
+const TEST_CONVERSATIONS: TestConversation[] = [
+  {
+    name: "Simple Math",
+    systemPrompt: "You are a helpful assistant. Be concise.",
+    userMessage: "What is 2*2? Just give the number.",
+    expectedInResponse: "4",
+  },
+  {
+    name: "Hello World",
+    systemPrompt: "You are a helpful assistant. Be very brief.",
+    userMessage: "Say hello",
+    expectedInResponse: "hello",
+  },
+  {
+    name: "Yes/No Question",
+    systemPrompt: "You are a helpful assistant. Answer only yes or no.",
+    userMessage: "Is Paris the capital of France?",
+    expectedInResponse: "yes",
+  },
+];
+
+async function runTest(
+  config: TestConfig,
+  conversation: TestConversation
+): Promise<void> {
+  console.log(
+    `- Testing ${config.provider} - ${config.modelId} - T=${config.temperature} - R=${config.reasoningEffort} - ${conversation.name}`
+  );
+
+  try {
+    let llm;
+    switch (config.provider) {
+      case "anthropic":
+        llm = new AnthropicLLM({
```

Comment:
We can use `getLLM()` !

I think the idea is, for any new model using the router, we add it in the provider whitelist and we run the test